src/Razor/src/Microsoft.AspNetCore.Razor.Language/TagHelperDescriptorExtensions.cs

```diff
@@ -38,5 +38,24 @@ public static bool KindUsesDefaultTagHelperRuntime(this TagHelperDescriptor tagH
             tagHelper.Metadata.TryGetValue(TagHelperMetadata.Runtime.Name, out var value);
             return string.Equals(TagHelperConventions.DefaultKind, value, StringComparison.Ordinal);
         }
+
+        public static bool IsRazorComponentTagHelper(this TagHelperDescriptor tagHelper)
```

Comment:
We already have an `IsComponentTagHelper` method but that also returns `true` for directive attributes so could not use that. I don't love the name I picked `IsRazorComponentTagHelper` so if you have a better one feel free to suggest.

Reply:
> We already have an IsComponentTagHelper method but that also returns true for directive attributes

Are you sure? That is not expected.

Reply:
Yeah that's not true bro. Look at my PR for `@attributes` where I added a different extension method that *includes* directive attributes.